### DISCUSSION TITLE
Need an explicit notification for PR comments

### DISCUSSION THREAD

**Discussion by chriskuech** [community/community#64225] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

My notifications are a mess. The filtering capabilities don't align at all with how I would want to filter out notifications. The existing categories--"author, ci activity, state change, review requested"--are so broad as to be unhelpful.

* I need to be able to see on the notifications feed that someone
  * specifically mentioned (`@`-ed) me in a comment
  * replied to a comment I made
  * commented on a PR I authored
* I need to not receive 500 other notifications just because someone pushed a commit or had a PR check fail

In current state, it's safe to assume someone will not see a comment unless they are the PR owner and it's blocking their PR merge.

**Comment by Jaber-Saed** [comment]
It sounds like you're dealing with overwhelming notifications in your workflow. GitHub's notification system might not provide the fine-grained control you're looking for, but here are some strategies you can consider to manage your notifications more effectively:

1. **Custom Filters**: Although GitHub's notification filtering options might not cover all your needs, you can make the most of the available filters. For instance, you can filter by repository or organization, which might help you focus on the notifications most relevant to you.

2. **Notification Settings**: Review and adjust your GitHub notification settings to align more closely with your preferences. You can manage notification methods (email, web, mobile), frequency, and notification types. While this won't provide extremely granular filtering, it might help you reduce the noise.

3. **Use Mention Notifications**: GitHub's mention system is designed to notify you when you're specifically mentioned using `@-username` in comments. Encourage your collaborators to use this approach when they need your direct attention.

4. **Subscribe to Specific Threads**: If you're involved in a specific thread (such as a PR or issue), you can manually subscribe to it. This way, you'll receive notifications for updates in that thread.

5. **Third-Party Integrations**: There are third-party tools and services that offer more advanced notification management for GitHub. Some tools can help you filter and manage notifications based on specific criteria.

6. **Issue/PR Management**: Keep a structured workflow for the issues and PRs you're involved in. For example, you might request that discussion happens primarily in issues or PRs and avoid commenting on unrelated code changes.

7. **Labels and Mentions**: Encourage your team to use labels to categorize issues and PRs. This way, you can filter notifications based on labels that are most relevant to you. Similarly, using @-mentions in comments can ensure the right people get notified when needed.

8. **Prioritization**: Create a daily routine where you dedicate specific time to go through your notifications. Prioritize the ones that require your immediate attention and leave the rest for later.

9. **Feedback to GitHub**: If you find that the existing notification system doesn't cater to your needs, consider providing feedback to GitHub. They might consider adding more granular controls based on user input.

Remember that effectively managing notifications requires a mix of setting expectations with your collaborators, utilizing available features, and possibly adapting your workflow. It's a good practice to regularly review and adjust your approach as your projects evolve and your needs change.

**Comment by chriskuech** [comment]
@Jaber-Saed sorry but GitHub notifications just needs to get 10x better. None of these suggestions seem applicable--

1. Custom Filters are not very useful as GitHub appears to heavily overload the "reason", thus the original feedback.
2. I am only referring to the Notifications view, not emails. These have been turned off already.
3. This relies on other users and is therefore unreliable.
4. This is predicated on having good initial signals to know there is a thread to subscribe to and therefore redundant.
5. Not sure what these are or how they would have more information than what GitHub can filter on.
6. We don't use this.
7. Same as 3.
8. This defeats the purpose of notifications (realtime notification).
9. This is what I did :)

I get a flood of notifications for PRs I'm automatically added as a reviewer as but are approved/closed by dependabot. I also get a flood of notifications if workflows fail with no way of distinguishing whether its an unimportant failure (CI check failed) or critical (deployment failure).

**Reply by tinder-tannerbennett** [reply]
Lmao @Jaber-Saed just took the question and pasted it into ChatGPT

**Comment by tinder-tannerbennett** [comment]
I have all the same issues as you. GitHub notifications need a complete revamp it seems. You hit the nail on the head; _none_ of the filters are meaningful. Hell, I have like 640 notifications in my inbox, and "participating" brings it down to... 620.

I don't know what makes those 3% of my notifications special, but that seems like more of a fluke than anything. I don't get a notification unless I have either commented/left a review or my review was requested, so I don't know what those 20 notifications could possibly be for.

My main gripe is, like you said, that I need a way to follow up on PRs I've left comments on, or comments where I was @ mentioned. Ideally I should just be able to see comment replies in their own filter though, instead of checking on every thread I've ever commented on to see if there are new replies.

**Reply by NSExceptional** [reply]
I actually needed to switch to my personal account, otherwise if this gets answered or I get a reply, it will get lost on my work account. I am @tinder-tannerbennett.

**Reply by NSExceptional** [reply]
@stephaniegiang could you please help with this? I saw you comment on a similar thread elsewhere
